packages/dso-toolkit/src/components/document-component/document-component.stories-of.ts

```diff
@@ -8,25 +8,31 @@ import {
   DocumentComponentArgs,
 } from "./document-component.args.js";
 import { imroContent } from "./document-component.content.js";
-import { DocumentComponent } from "./document-component.models.js";
+import { DocumentComponent, DocumentComponentMode } from "./document-component.models.js";
 
 import { StoriesParameters, StoryObj } from "../../template-container";
 import { compiler } from "markdown-to-jsx";
 import { MetaOptions } from "../../storybook/meta-options.interface";
 
 export type DocumentComponentDecorator<TemplateFnReturnType> = (story: PartialStoryFn) => TemplateFnReturnType;
 
+type DocumentComponentStoryDemo = StoryObj<
+  {
+    jsonFile: string;
+    openDefault: boolean;
+    showCanvas: boolean;
+    mode: DocumentComponentMode;
+    href: string;
+    ozonContentAnchorClick: HandlerFunction;
+    tableOfContentsClick: HandlerFunction;
+  },
+  Renderer
+>;
+
 interface DocumentComponentStories {
   Default: StoryObj<DocumentComponentArgs, Renderer>;
-  Demo: StoryObj<
-    {
-      jsonFile: string;
-      openDefault: boolean;
-      showCanvas: boolean;
-      ozonContentAnchorClick: HandlerFunction;
-    },
-    Renderer
-  >;
+  Demo: DocumentComponentStoryDemo;
+  DemoInhoudsopgave: DocumentComponentStoryDemo;
```

Comment:
Klopt dit? Er is nu één demo StoryObj met properties die niet voor beide demo's gelden?

Reply:
Hmmm, ik heb het nu zo dat je in beide demo's met een control kan schakelen met tussen mode document en mode table-of-contents. Wat wil je dan?

Reply:
Ik resolve deze, deze is outdated en heb wat meer context nodig voor het antwoord